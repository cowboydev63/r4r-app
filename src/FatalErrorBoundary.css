.r4r-boundary {
    height: 100%;
    width: 100%;

}

.r4r-boundary.r4r-DEFAULT {
    display: flex;
    justify-content: center;
    align-items: center;
    background: rgb(226, 226, 226);
}

.boundary__inner.r4r-DEFAULT {
    display: flex;
    flex-flow: column nowrap;
    justify-content: center;
    align-items: center;
    background: white;
    box-shadow: 0 15px 15px -12px rgba(0,0,0,0.3);
    padding: 15px;
}

.r4r-boundary.r4r-DEFAULT p {
    margin: 10px;
}