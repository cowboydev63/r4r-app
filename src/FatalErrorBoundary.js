import React from 'react';
import PropTypes from 'prop-types';
import Error from './components/Error';
import { registerError } from './state/error/actions';
import './FatalErrorBoundary.css';

class FatalErrorBoundary extends React.Component {
    state = {
        hasError: false,
    }

    static propTypes = {
        dispatch: PropTypes.func.isRequired,
    }

    componentDidCatch(error, info){
        this.setState({
            hasError: true,
        })
        this.props.dispatch(registerError('FATAL'));
    }

    render() {
        return (
            this.state.hasError ? <Error /> : this.props.children
        )
    }
}

export default FatalErrorBoundary;