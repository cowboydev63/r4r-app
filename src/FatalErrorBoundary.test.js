import React from 'react';
import { shallow } from 'enzyme';
import FatalErrorBoundary from './FatalErrorBoundary';

describe('FatalErrorBoundary HOC', () => {
  it('renders without crashing', () => {
    shallow(<FatalErrorBoundary />);
  });

  it('renders correctly with no error', () => {
    const wrapper = shallow(<FatalErrorBoundary />);
    expect(wrapper).toMatchSnapshot();
  })
})