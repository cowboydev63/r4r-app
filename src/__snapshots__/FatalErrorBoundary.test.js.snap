// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FatalErrorBoundary HOC renders correctly with no error 1`] = `""`;
